### PR TITLE
Fix: #6684 DataTable column resize on touchscreens

### LINKED ISSUE #6684: DataTable: Resize columns on touch device
### Describe the bug

When I try to resize datatable columns on a touch device, nothing happens, I just horizontally scroll through the table.
It's probably hooked up to mouseEvents, instead of pointerEvents.
There might be a good reason for this not working, if so, I'd suggest taking this up in the docs.

No stackblitz attached, because it's reproducible on the docs on a touch device as well:
https://primereact.org/datatable/#column_resize

### Reproducer

_No response_

### PrimeReact version

10.6.5

### React version

18.x

### Language

TypeScript

### Build / Runtime

Create React App (CRA)

### Browser(s)

_No response_

### Steps to reproduce the behavior

_No response_

### Expected behavior

_No response_

### PR DESCRIPTION
**Pull request by KirilCycle**
### Defect Fixes
Fix #6684

Handle touch events

### CODE REVIEW

File: components/lib/datatable/DataTable.js [hunk 1]
```diff
@@ -105,6 +105,25 @@ export const DataTable = React.forwardRef((inProps, ref) => {
...
+            if (columnResizing.current) {
+                onColumnResize(event);
+            }
+        }
+    });
+
+    const [bindDocumentTouchEndListener, unbindDocumentTouchEndListener] = useEventListener({
+        type: 'touchend',
+        listener: () => {
+            if (columnResizing.current) {
```
Comment (KirilCycle) [inline components/lib/datatable/DataTable.js:120]: These conditions are the same as when binding mouse events, should they be refactored?

File: components/lib/datatable/DataTable.js [hunk 2]
```diff
@@ -522,9 +541,16 @@ export const DataTable = React.forwardRef((inProps, ref) => {
...
         resizeColumn.current = column;
+
         resizeColumnElement.current = event.currentTarget.parentElement;
+
         columnResizing.current = true;
-        lastResizeHelperX.current = event.pageX - containerLeft + elementRef.current.scrollLeft;
+
+        if (event.type == 'touchstart') {
+            lastResizeHelperX.current = event.changedTouches[0].clientX - containerLeft + elementRef.current.scrollLeft;
+        } else {
```
Comment (melloware) [inline components/lib/datatable/DataTable.js:551]:
```js
lastResizeHelperX.current = (event.type === 'touchstart' 
    ? event.changedTouches[0].clientX : event.pageX) - containerLeft + elementRef.current.scrollLeft;
```

File: components/lib/datatable/DataTable.js [hunk 3]
```diff
@@ -536,7 +562,12 @@ export const DataTable = React.forwardRef((inProps, ref) => {
 
         resizeHelperRef.current.style.height = elementRef.current.offsetHeight + 'px';
         resizeHelperRef.current.style.top = 0 + 'px';
-        resizeHelperRef.current.style.left = event.pageX - containerLeft + elementRef.current.scrollLeft + 'px';
+
+        if (event.type == 'touchmove') {
+            resizeHelperRef.current.style.left = event.changedTouches[0].clientX - containerLeft + elementRef.current.scrollLeft + 'px';
+        } else {
+            resizeHelperRef.current.style.left = event.pageX - containerLeft + elementRef.current.scrollLeft + 'px';
+        }
```
Comment (melloware) [inline components/lib/datatable/DataTable.js:570]:
```suggestion
resizeHelperRef.current.style.left = (event.type === 'touchstart' 
    ? event.changedTouches[0].clientX : event.pageX) - containerLeft + elementRef.current.scrollLeft+ 'px';
```

**Review by melloware — CHANGES_REQUESTED** [review]

File: components/lib/datatable/DataTable.js (hunk 1 above)
Comment (melloware) [reply components/lib/datatable/DataTable.js:120]: yes

File: components/lib/datatable/DataTable.js [hunk 4]
```diff
@@ -86,23 +86,28 @@ export const DataTable = React.forwardRef((inProps, ref) => {
         setD_rowsState(props.rows);
     }
 
+    const moveListener = (event) => columnResizing.current && onColumnResize(event);
```
Comment (melloware) [inline components/lib/datatable/DataTable.js:89]: columnResizeStartListener and columnResizeEndListener would be better names.

**Review by melloware — CHANGES_REQUESTED** [review]

**Comment by Josje92** [comment]
@KirilCycle is there are reason you didn't go with pointer-events instead of adding the touch-events?
As far as [my info](https://medium.com/@elmarti/farewell-mouse-touch-events-welcome-pointer-events-db36cee48a8a) goes, implementing pointerevents would reduce a couple of lines of code and behave the same way.

Then again, I haven't been contributing to this project yet, so I might not oversee the risks. Just wanted to (maybe) be of help.

**Comment by Josje92** [comment]
@melloware This fix would help me and my project greatly, any chance you'll be able to look into it for @KirilCycle (/ and me)? 😄 
Please

**Comment by melloware** [comment]
merged!

**Pull request merged**